server/app/models/grid_service_deploy.rb

```diff
@@ -36,6 +36,7 @@ class GridServiceDeploy
   index({ created_at: 1 }, { background: true })
   index({ queued_at: 1 }, { background: true })
   index({ started_at: 1 }, { background: true })
+  index({ finished_at: 1 }, { background: true })
```

Comment:
The `finished_at` index change probably doesn't belong in this PR... it seems to need some more complex compound index per #3046.